lib/Requestable.js

```diff
@@ -250,13 +250,7 @@ class Requestable {
 
             response.data = results;
             return response;
-         }).catch(function(error) {
-            if (cb) {
-               cb(error);
-            } else {
-               throw error;
-            }
-         });
+         }).catch(callbackErrorOrThrow(cb, path));
```

Comment:
Looking at this again, I think you were on to something with this change in the first place, I think that perhaps it needed to be a bit more nuanced. Maybe something like:

```javascript
.catch(function(error) {
  // we haven't processed this error yet
  if (!(error instanceof ResponseError)) {
    callbackErrorOrThrow(cb, path)(error);

  // we have already processed this error
  } else if (cb) {
    cb(error);
  } else {
    throw error
  }
});
```

Thoughts?